RosNet/RosMessageParser/MessageDataParser.cs

```diff
@@ -1,99 +1,95 @@
-                posBytes += 4;
-            }
+                ArrayFieldValue arrayFieldValue = ParseArrayDefinitionField(definitionFieldValue as ArrayFieldValue, messageData, ref posBytes);
+            }
             else
             {
-                arrayLength = arrayDefinitionField.FixedArrayLength;
-            }
-
-            
-            int j = 0; //position in arrayDefinitionField
-            int arrayDefinitionLength = arrayDefinitionField.ArrayFields.Count;
-            var arrayFields = new List<FieldValue>();
-            for (uint i = 0; i < arrayLength; i++)
-            {
-                var definitionFieldValue = arrayDefinitionField.ArrayFields[j];
-                if (definitionFieldValue is ArrayFieldValue)
-                {
-                    ArrayFieldValue arrayFieldValue = ParseArrayDefinitionField(definitionFieldValue as ArrayFieldValue, messageData, ref posBytes);
-                }
-                else
-                {
-                    FieldValue field = ParseDefinitionField(definitionFieldValue, messageData, ref posBytes);
-                    arrayFields.Add(field);
+                FieldValue field = ParseDefinitionField(definitionFieldValue, messageData, ref posBytes);
+                arrayFields.Add(field);
 
-                    j++;
+                j++;
 
-                    if (j == arrayDefinitionLength) //starts at beginning of arrayDefinitionField if it reaches the end
-                    {
-                        j = 0;
-                    }
+                if (j == arrayDefinitionLength) //starts at beginning of arrayDefinitionField if it reaches the end
+                {
+                    j = 0;
                 }
             }
-            return new ArrayFieldValue(arrayDefinitionField.Name, arrayFields, arrayDefinitionField.DataType);
         }
+        return new ArrayFieldValue(arrayDefinitionField.Name, arrayFields, arrayDefinitionField.DataType);
+    }
 
-        /// <summary>
-        /// Parses a field in the messageData on the position posBytes
-        /// </summary>
-        /// <returns> The parsed fieldvalue</returns>
-        public static FieldValue ParseDefinitionField(FieldValue definitionField, byte[] messageData, ref int posBytes)
-        {
-            byte[] value = messageData.Skip(posBytes).Take(definitionField.GetByteLength()).ToArray();
-            var field = new FieldValue(definitionField.Name, definitionField.DataType, value);
+    /// <summary>
+    /// Parses a field in the messageData on the position posBytes
+    /// </summary>
+    /// <returns> The parsed fieldvalue</returns>
+    public static FieldValue ParseDefinitionField(FieldValue definitionField, byte[] messageData, ref int posBytes)
+    {
+        byte[] value = messageData.Skip(posBytes).Take(definitionField.GetByteLength()).ToArray();
+        var field = new FieldValue(definitionField.Name, definitionField.DataType, value);
```

Comment:
This is wrong if the type is a string, because you are then only reading the length of the string but not the string itself